app/models/user.rb

```diff
@@ -41,9 +41,23 @@ class User < ApplicationRecord
 
   has_many :sessions, class_name: 'Authie::Session', as: :user, dependent: :destroy
 
-  NAME_REGEX = /\A[A-Za-z0-9_]+\z/.freeze
+  NAME_REGEX     = /\A[A-Za-z0-9_]+\z/.freeze
+  PASSWORD_REGEX = /\A.{8,}+\z/.freeze
+
+  # Turn off default password validations because they include a presence
+  # validation, which we don't require as some accounts can be link-only. But
+  # turn on the other validations manually -- validating that `password` and
+  # `password_confirmation` match when both supplied, and max length.
+  #
+  # Although it's theoretically bad to have max lengths on passwords, BCrypt
+  # truncates at 72 bytes, so we should enforce that here rather than let users
+  # pass in 73 bytes and be surprised that we only use the first 72.
+  has_secure_password(validations: false)
+  validates_confirmation_of :password, allow_blank: true
+  validates_length_of       :password, maximum: ActiveModel::SecurePassword::MAX_PASSWORD_LENGTH_ALLOWED
 
   validates :name, presence: true, uniqueness: true, format: {with: NAME_REGEX}
+  validates :email, presence: true, uniqueness: true # TODO: Add a not-null to database
```

Comment:
Yeah, it's because I wanted to deploy this branch to beta to test mailers, but when it's deployed to beta migrations are run against the production database. So if this column was marked not null in the production database before the branch was deployed to production, production rails would hit errors about trying to insert nulls.

Reply:
Makes perfect sense, thanks for the explanation.